### DISCUSSION TITLE
Error when trying to add revit_analyze_electric_tariff in workflow

### DISCUSSION THREAD

**Discussion by devrtg** [chrisbalbach/chrisbalbach.github.io#28] [Q&A]
Hello Chris, I've watched your presentation on AU 23 and now I'm trying to recreate then workflows on my machine.
I've downloaded the workflows and measures from the repo and dragged into my folder. Then when trying to run, it fails... The error appears to be:
 
[10:46:58.757186 ERROR] uninitialized constant RevitAnalyzeElectricTariff::OsLib_HelperMethods
	eval:368:in `const_missing'

I am a revit api developer and have a good knowledge in C#, but unfamiliar with Ruby. After searching I've discovered that OsLib_HelperMethods is part of the OpenStudio API. I already downloaded the SDK but still without success... Can you point me in the right direction? Amazing presentation btw!

I've attached my run.log file
[run.log](https://github.com/user-attachments/files/17477375/run.log)

Greetings from Brazil
Ricardo

**Comment by devrtg** [comment]
Actually, I've solved the problem just by adding
require "openstudio/extension/core/os_lib_helper_methods"

In the begining of the measure... My bad hehe

**Comment by chrisbalbach** [comment]
Glad to hear you worked it out.

Those measures were very kludgy but they did work!

Also wanted to make sure you were aware of the presentation given last week at AU 2024- there is an updated openstudio measure based workflow wrapped in a user installed add on.

You may be interested in the measures that install with the add on - they have been significantly extended. They allow hvac systems to be modeled, with overrides.

Also, the add on requires revit 2025 but the measures only need openstudio version 3.6

On the next few weeks we will be updating the add on to handle non electric hvac systems, the ashtray baseline., and service hot water system  modeling capabilities.

Thanks for your interest in the project!

Chris

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
